hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/RatisPipelineUtils.java

```diff
@@ -100,4 +105,45 @@ static void destroyPipeline(DatanodeDetails dn, PipelineID pipelineID,
           true, p.getId());
     }
   }
+
+  static int encodeNodeIdsOfFactorThreePipeline(List<DatanodeDetails> nodes) {
+    if (nodes.size() != HddsProtos.ReplicationFactor.THREE.getNumber()) {
+      return 0;
+    }
+    List<UUID> nodeIds = nodes.stream()
+        .map(DatanodeDetails::getUuid).distinct()
+        .collect(Collectors.toList());
+    nodeIds.sort(new ComparableComparator());
+    // Only for Factor THREE pipeline.
+    return new HashCodeBuilder()
+        .append(nodeIds.get(0).toString())
+        .append(nodeIds.get(1).toString())
+        .append(nodeIds.get(2).toString())
+        .toHashCode();
+  }
+
+  /**
+   * Return first existed pipeline which share the same set of datanodes
+   * with the input pipeline.
+   * @param stateManager PipelineStateManager
+   * @param pipeline input pipeline
+   * @return first matched pipeline
+   */
+  static Pipeline checkPipelineContainSameDatanodes(
```

Comment:
Should we move this to a non-static PipelineStateManager method?

Reply:
I put this into Utils for all classes to access. I tried PipelineStateManager first and Sammi advised to put it into Utils for some test classes who don't have pipelineStateManager initialized.

Reply:
Both should work just non-static seems cleaner. But if there are too much additional work to fix the unit tests. I'm OK with current approach.